### ISSUE TITLE
Implement PyPlot

### ISSUE THREAD

**Issue by Evizero** [tbreloff/Plots.jl#5]
I've read the ANN on the mailing list.

You wrote that you might want to implement Vega.jl next. Could you maybe consider doing PyPlot instead of Vega next?

Here is why I ask: My usual usecases for plotting are either for analysis (Gadfly), or simulation (UnicodePlot / PyPlot). Pyplot simply because it opens a dedicated window where I can have live plots etc, which are useful in machine learning to keep an eye on the learning curve.

So basically I would like to get the Ball rolling by incorporating this library into all my ML Efforts (which for the moment are SupervisedLearning.jl , KSVM.jl and soon FeatureSelection.jl) as the Base plotting library where I provide custom plotting functions. For example to plot the decision boundary, margin, and support vectors for fitted SVM models.

**Comment by tbreloff** [comment]
It's great to hear you'll start basing some work off of this.  I hesitate with putting PyPlot at the top of the list, mainly because Qwt is so similar to it.  You should give Qwt a try if you haven't already... it will open a dedicated window and you can do updating/interactive plotting with pan/zoom, etc.  Installation is fairly straightforward (especially on Mac, as you can just "brew install" pyqt and pyqwt before cloning Qwt.jl).  Give it a try and see if that meets your needs, and please let me know if there's important functionality that's missing.  I have Vega.jl high on my list because I feel like many people want the D3 backend, and Vega seems pretty cool, and I'd like UnicodePlots.jl so there is an option with minimal dependencies.  Let me know your thoughts... I can be convinced...

**Comment by Evizero** [comment]
I have to admit I have been ignorant to other packages that do similar things to PyPlot. I'll check Qwt out. Chances are - based on what you wrote - it will be exactly what I need anyway

**Comment by tbreloff** [comment]
After a few grueling hours, PyPlot is now mostly supported in Plots.  Check out the examples page.  No subplots, and I'm a little disappointed in how much space the legends take up.  Also I was having trouble figuring out how to avoid showing the interactive window with `show=false`.  I now remember why I swore off matplotlib years ago... 

cc: @stevengj

**Comment by Evizero** [comment]
Cool! Thanks for all your efforts! I appreciate you spending so much time on Julia

Yea, I am not rushing to matplotlib's defence in that regard. It's more of a love-hate relationship in my case

**Comment by tbreloff** [comment]
Closing this in favor of #9, since that's all that is left.
